# Getting 2010 U.S. Census Data

### To use their API, go [here](https://www.census.gov/developers/) and click on the "Request a key" link

### Important links / resources:
- [List](https://api.census.gov/data.html) of all available data
- API user's [guide](https://www.census.gov/data/developers/guidance/api-user-guide.html)
- FIPS code [location](https://www2.census.gov/geo/docs/reference/codes/files)
- Geographic codes [reference](https://www.census.gov/geo/reference/geocodes.html)
- State level URL [formats](https://api.census.gov/data/2010/sf1/geo.html)
- Summary File 1 (SF1) [reference](https://api.census.gov/data/2010/sf1.html)

**NOTE:** For Ohio, FIPS state code = 39

- Quick census [summary](https://www.census.gov/2010census/popmap/ipmtext.php?fl=39) for Ohio

### U.S. Census do not have "Hispanic or Latino" as a separate race designation since it is an ethnicity

- To obtain "White, but not Hispanic or Latino", I obtained the "P0050003" record.  To obtain "Hispanic or Latino", I obtained the "P0050011" record.

In [90]:
import requests

key = 'your_api_key'

data_by_race = 'P0030001,P0050003,P0050011,P0030003,P0030004,P0030005,P0030006,P0030007,P0030008'

url = "https://api.census.gov/data/2010/sf1?get=" + data_by_race + ",NAME&for=state:39&key=" + key
url_county = "https://api.census.gov/data/2010/sf1?get=" + data_by_race + ",NAME&for=county:*&in=state:39&key=" + key

r = requests.get(url)
r_county = requests.get(url_county)
print(r.status_code)
print(r_county.status_code)

200
200


### Ohio's raw json data:

In [84]:
r.json()

[['P0030001',
  'P0050003',
  'P0050011',
  'P0030003',
  'P0030004',
  'P0030005',
  'P0030006',
  'P0030007',
  'P0030008',
  'NAME',
  'state'],
 ['11536504',
  '9359263',
  '180174',
  '1407681',
  '25292',
  '192233',
  '4066',
  '130030',
  '237765',
  'Ohio',
  '39']]

### Ohio's raw json data by county:

In [91]:
r_county.json()[:2]

[['P0030001',
  'P0050003',
  'P0050011',
  'P0030003',
  'P0030004',
  'P0030005',
  'P0030006',
  'P0030007',
  'P0030008',
  'NAME',
  'state',
  'county'],
 ['28550',
  '27713',
  '169',
  '87',
  '108',
  '33',
  '2',
  '59',
  '379',
  'Adams County',
  '39',
  '001']]

### Print out of populations by race:

In [74]:
print("Total population:", r.json()[1][0])
print("White, but not Hispanic or Latino:", r.json()[1][1])
print("Hispanic or Latino:", r.json()[1][2])
print("African American:", r.json()[1][3])
print("American Indian/Alaskan Native:", r.json()[1][4])
print("Asian:", r.json()[1][5])
print("Native Hawaiian/Pacific Islander:", r.json()[1][6])
print("Some other race:", r.json()[1][7])
print("Two or more races:", r.json()[1][8])

Total population: 11536504
White, but not Hispanic or Latino: 9359263
Hispanic or Latino: 180174
African American: 1407681
American Indian/Alaskan Native: 25292
Asian: 192233
Native Hawaiian/Pacific Islander: 4066
Some other race: 130030
Two or more races: 237765


### Let's double-check that all the population figures add up to the total population total of 11,536,504:

In [76]:
int(r.json()[1][1])+int(r.json()[1][2])+int(r.json()[1][3])+int(r.json()[1][4])+int(r.json()[1][5])+ \
int(r.json()[1][6])+int(r.json()[1][7])+int(r.json()[1][8])

11536504

We can also check these numbers against this [summary](https://www.census.gov/2010census/popmap/ipmtext.php?fl=39) for Ohio.

### Coming up: make a function to get population data for any state

**An example function could look like:**<br>
```def state_census_by_race(key, state, as_perc):
        define function...
```